nixos/modules/services/networking/kea.nix

```
@@ -288,9 +289,9 @@ in
         "kea-dhcp-ddns-server.service"
       ];
 
-      environment = {
-        KEA_PIDFILE_DIR = "/run/kea";
-        KEA_LOCKFILE_DIR = "/run/kea";
+      environment = rec {
+        KEA_PIDFILE_DIR = "/run/${config.systemd.services.kea-ctrl-agent.serviceConfig.RuntimeDirectory}";
+        KEA_LOCKFILE_DIR = KEA_PIDFILE_DIR;
       };
```

I think making that recursive and concating things makes it less readable.

```suggestion
      environment = rec {
        KEA_PIDFILE_DIR = "/run/kea-ctrl-agent";
        KEA_LOCKFILE_DIR = "/run/kea-ctrl-agent";
      };
```